### PR TITLE
Issue #471: better handling of sura/ayah urls

### PR DESCRIPTION
**Pull request by atalebagha**
## Related Issue
https://github.com/quran/quran.com-frontend/issues/471
## Changelog
- Routes with colons `/39:4` are rerouted to `/39/4`
- Inverted ranges are fixed `/39/5-4` fixed to `/39/4-5`
- Url with a range of 1 just prints 1 ayah `/39:5-5` fixes to `/39/5`

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32852

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32853

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32854

**Comment by hassansin** [comment]
Quick thought: for handling redirection, you could replace these lines (https://github.com/quran/quran.com-frontend/blob/master/src/routes.js#L27-L31) with built-in `Redirect` component like `<Redirect from="/:surahId:(:range)" to="/:surahId(/:range)" />` . then in `isValidSurah` you only need to handle the inverted range

**Comment by atalebagha** [comment]
Thanks, will do!

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32855

**Comment by sabeurthabti** [comment]
LGTM

**Comment by sabeurthabti** [comment]
@atalebagha Thank you bro!

File: src/utils/checkValidSurah.js [hunk 1]
```diff
@@ -4,4 +4,16 @@ export default function isValidSurah(nextState, replaceState) {
   if (isNaN(surahId) || surahId > 114 || surahId < 1) {
     replaceState('/error/invalid-surah');
   }
+
+  if (nextState.params.range) {
```
Comment (sabeurthabti) [inline src/utils/checkValidSurah.js]: I would love for this to be unit tested, inshallah in the future.

**Pull request merged**